src/codeCache.cpp

```diff
@@ -311,3 +307,60 @@ size_t CodeCache::usedMemory() {
     }
     return bytes;
 }
+
+bool CodeCache::isValidHandle(void* handle) const {
+#ifdef __linux__
```

Comment:
CodeCache is just a data structure, it's not good to have platform-dependent code here.
`dlopen`, `dlinfo` and even more so, mapping-related stuff are more suitable for `Symbols` class, whose implementation is platform-dependent. Moreover, in your implementation, that responsibility is already split across CodeCache and Symbols: the code calls `Symbols::isMainExecutable` and `Symbols::isLoader` which should not be public functions in the first place, since these are platform-specific implementation details.